### PR TITLE
Feat/price quantity base commission

### PR DESCRIPTION
**Pull request by Aunshon**
Adding three more commission types.
Product Quantity and Product Price and Vendor Sale

There is also a fix in a refund, Also in previous Dokan, the issue exists. If you order a product, complete the order, delete the product and then try to refund the order the admin commission will be zero. It should not be zero, commission should be as applied to the ordered item.

**Note: This PR has a dependency with Dokan-pro [PR](https://github.com/weDevsOfficial/dokan-pro/pull/1577)**

[Video](https://www.awesomescreenshot.com/video/6630875?key=54f322ada4655d78752f8571cb83a279)

### CODE REVIEW

**Comment by shanjidah** [comment]
**Scenario1:** Deleted all commission types as Product price, shows console error.

**Here is the screenshot:**

Screen Shot 2021-12-30 at 12 23 52 PM

**Comment by shanjidah** [comment]
Scenario2: Commission Label is not matched with Commission types Functionalities.

Screen Shot 2021-12-30 at 12 30 47 PM

**Comment by shanjidah** [comment]
Scenario 3: Commission type is not working on Vendor Profile.

Screen Shot 2021-12-30 at 4 58 16 PM

**Review by tanmayjay — CHANGES_REQUESTED** [review]

File: src/admin/components/Fields/SinglePriceQuantityVendorSale.vue [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            }
+
+            this.allCommission[index].from > Number(e) && '' != e ? this.showFromErrorMessage( index, this.allCommission[index].from, e ) : this.hideFromErrorMessage( index );
+            
+            await this.updateCommissionValue( e, 'to', index, 'number' );
+            await this.generateNextRow( e, index );
+        },
+
+        showFromErrorMessage( index, from, to ) {
+            this.$refs[`dokan-${this.selectedCommissionLabel}-from-msg${index}`][0].innerText = `Must be more or equal ${from}`;
```
Comment (tanmayjay) [inline src/admin/components/Fields/SinglePriceQuantityVendorSale.vue:102]: The text should be translatable. And the sentence will be `Must be more than or equal to `

File: src/admin/pages/VendorPaymentFields.vue [hunk 2]
```diff
@@ -222,8 +259,112 @@ export default {
...
+
+        getAndCheckCommissionType( begin = false, value = '' ) {
+            let selectedCommissionName = this.selectedCommissionType.name;
+            let commissionData = {...this.newCommissionData};
+
+            value ? commissionData.from = value : '';
+
+            if ( this.newCommissions.includes( selectedCommissionName ) ) {
+                return begin ? [commissionData] : commissionData;
+            } else {
```
Comment (tanmayjay) [inline src/admin/pages/VendorPaymentFields.vue:274]: `else` is not needed here. Just write `return ''`

File: src/admin/pages/VendorPaymentFields.vue [hunk 3]
```diff
@@ -222,8 +259,112 @@ export default {
...
+        },
+
+        get_vendor_commission_array( commissions ) {
+            let all_commissions = [];
+            all_commissions = 'string' === typeof commissions ? [] : commissions;
+
+            return all_commissions;
+        },
+
+        // async generateNextRow( data ) {
```
Comment (tanmayjay) [inline src/admin/pages/VendorPaymentFields.vue:322]: Remove this if not needed.

File: templates/settings/new-commission.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ * Dokan new commission type ui ( Product price / Vendor sale / Product quantity )
+ *
+ * @since DOKAN_LITE_SINCE
+ *
+ * @package dokan
+ */
+
+$admin_commissions    = get_user_meta( $args->ID, 'dokan_admin_percentage', true );
+$currency_symbol      = get_woocommerce_currency_symbol();
+$commission_i18n_data = [
```
Comment (tanmayjay) [inline templates/settings/new-commission.php:12]: Why do you need this? You can use the values directly inside html.

File: templates/settings/new-commission.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+<tr class="new_commission_body dokan-new-commission dokan-hide">
+    <th><?php esc_html_e( 'Admin Commission ', 'dokan' ); ?></th>
+    <td class="new_commission_fields_body " id="new_commission_fields_body">
+        <fieldset class="dokan-new-commission-wraper">
+            <div class="dokan-new-commission-header dokan-new-commission-row">
+                <div class="dokan-new-commission-col-from"><?php esc_html_e( $commission_i18n_data['from'] ); ?></div>
+                <div class="dokan-new-commission-col-to"><?php esc_html_e( $commission_i18n_data['to'] ); ?></div>
+                <div class="dokan-new-commission-col-ct"><?php esc_html_e( $commission_i18n_data['type'] ); ?></div>
+                <div class="dokan-new-commission-col-commission"><?php esc_html_e( $commission_i18n_data['commission'] ); ?></div>
```
Comment (tanmayjay) [inline templates/settings/new-commission.php:33]: text domain is missing for all the `esc_html_e`

File: templates/settings/new-commission.php [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        </fieldset>
+    </td>
+</tr>
+
+<script>
+    var dokan_commission_i18n_data = <?php echo json_encode( $commission_i18n_data ); ?>;
+</script>
+
+<?php
+if ( ! empty( get_current_screen()->id ) && 'user-edit' === get_current_screen()->id ) {
```
Comment (tanmayjay) [inline templates/settings/new-commission.php:82]: Remove this from here and use this inside another relevant file.

Comment (tanmayjay) [reply templates/settings/new-commission.php:82]: Also, from the line#78, do that using `wp_localize_script`

**Review by tanmayjay — CHANGES_REQUESTED** [review]

**Review by tanmayjay — CHANGES_REQUESTED** [review]

**Review by tanmayjay — APPROVED** [review]

**Comment by nurul-umbhiya** [comment]
Requirements have been updated. Need to provide a new pr.

**Pull request closed without merging**